src/hooks/useMovietrailer.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { API_OPTIONS } from "../utils/constants";
+import { addTrailerVideo } from "../utils/moviesSlice";
+
+interface VideoDetails {
+  id: string;
+  iso_639_1: string;
+  iso_3166_1: string;
+  key: string;
+  name: string;
+  official: boolean;
+  published_at: string;
+  site: string;
+  size: number;
+  type: string;
+}
+
+const useMovieTrailer = ({ movieId }: { movieId: number }) => {
+  const dispatch = useDispatch();
+
+  const getMovieVideo = async () => {
+    const data = await fetch(
+      `https://api.themoviedb.org/3/movie/${movieId}/videos`,
+      API_OPTIONS
+    );
+    const jsonData = await data.json();
+    const filterData = jsonData.results.filter(
+      (video: VideoDetails) => video.type === "Trailer"
```

Comment:
Trailer seems to be an important parameter which could affect the filtering, to avoid any mistyping create a type for it.

Check if there are other types and create the type for it to equate to

Reply:
Resolved